eservice/lib/libpdo_enclave/contract_request.h

```diff
@@ -40,15 +40,17 @@ class ContractRequest
     };
     Operation operation_; /* either "initialize" or "update" */
 
-    ContractResponse process_initialization_request(void);
-    ContractResponse process_update_request(void);
+    ContractResponse process_initialization_request(ContractState& contract_state);
+    ContractResponse process_update_request(ContractState& contract_state);
 
 public:
     std::string contract_id_;
     std::string creator_id_;
     ByteArray state_encryption_key_ = {};
+    ByteArray input_state_hash_ = {};
+    ByteArray contract_id_hash_ = {};
+    ByteArray code_hash_ = {};
```

Comment:
not sure why these are added to the fields in contract request. other than testing them for consistency, where do we use them?

Reply:
in contract_enclave.cpp, ~line 250
Before, these were given to the unpack (which constructed the ContractState) from inside ContractRequest.
Now that I pulled ContractState out, ContractRequest has to save them, so they can later be used to build ContractState.

Reply:
but the logic has changed... i need to go back and look at the delta... but the old code verified that the actual hash of the state is consistent with the one sent into the request. same with code & contractid. that security check seems to be lost.

Reply:
Yes, logic has changed, but checks have been maintained.
* The state hash and contract id are still verified by the unpack in ContractState. But now the unpack is not called by contractRequest, but rather by the ContractState constructor. Interesting point: I could make `unpack` and `initialize` private!
* the contract code has is still verified by ContractCode::FetchFromState. But this is not called from ContractRequest, but from ContractEnclave after ContractState is constructed